client/src/documentLinkProvider.ts

```diff
@@ -29,18 +32,64 @@ export class BitbakeDocumentLinkProvider implements vscode.DocumentLinkProvider
     */
     const filenames = linksData.map(link => link.value.split(';')[0])
     const filenamesRegex = '{' + filenames.join(',') + '}'
-    const parentDir = document.uri.path.split('/').slice(0, -1).join('/')
-    const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+    const parentDir = path.dirname(uri.path)
+    const pnDir = path.join(parentDir, extractRecipeName(uri.path) as string)
     const filesDir = path.join(parentDir, 'files')
-    return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
-      ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+
+    /**
+     * connman/
+     * ├── connman-gnome/
+     *     └── images/
+     * ├── connman-gnome_0.7.bb
+     * └── ... (other files and folders)
+     */
+    for (let i = 0; i < linksData.length; i++) {
```

Comment:
This might be very slow if the user has a huge number of files, however it's bad practice.
A more optimized way would be to spawn a single process that can list all directories and then handle the result in JS.
Either through spawn, but not portable, better would be to use an appropriate library.

Reply:
Check 1df4aeed6bd7ac71fb92c6c8e01a3ac140ef0b66